### PR TITLE
VP-1071 send emails on interest status changes

### PR DESCRIPTION
**Pull request by avowkind**
## I do solemnly swear that I have:
- [x] Run `npm test` and all the tests pass.
- [x] Run `npm run lint` and there are no warnings.
- [x] Not decreased the overall test coverage?
- [x] Included the Jira ID and description in the PR title

## Proposed Changes? 🤔
When a person indicates interest in a volunteering opportunity a new interest record is created and this tracks changes as they are invited, commit, withdraw etc.  Once the event is completed the record becomes an archivedInterest record and the state may become attended or notattended.
On each state change the controller publishes a TOPIC_INTEREST__UPDATED message.  This is picked up by the person.subscribe.js module which generates emails to the volunteer, the op requestor or both depending on the state change.  emails are provided with the details of the opportunity, requestor, volunteer and interest status. 

New templates added for each state. templates are of the form interest_vp_state or interest_op_state. 

## More information
Emails on interest state changes were already present but sent directly from the interest controller.  this now uses the pub/sub method to dispatch the change and handle it later which will speed up the request response.

### CODE REVIEW

File: server/api/person/__tests__/person.email.spec.js [hunk 1]
```diff
@@ -60,14 +59,14 @@ test('Send acknowledgeInterest email to person', async t => {
   t.regex(sentMail[0].subject, /Confirming your interest/)
 })
 
-test('render acknowledgeInterest email to person', async t => {
+test.only('render acknowledgeInterest email to person', async t => {
```
Comment (guttmann) [inline server/api/person/__tests__/person.email.spec.js]: Should this `.only` be here?

File: server/api/person/person.email.js [hunk 2]
```diff
@@ -21,8 +22,8 @@ module.exports.emailPerson = async (template, to, props, renderOnly = false) =>
     send: true,
     subjectPrefix: config.env === 'production' ? false : `[${config.env.toUpperCase()}] `,
 
-    // Comment the line below to see preview email
-    preview: false,
+    // uncomment the line below to see preview email
+    // preview: true,
```
Comment (guttmann) [inline server/api/person/person.email.js]: Is leaving `preview` commented out equivalent to having `preview: false`?

File: server/api/person/__tests__/person.email.spec.js (hunk 1 above)
Comment (avowkind) [reply server/api/person/__tests__/person.email.spec.js]: good catch.

File: server/api/person/person.email.js (hunk 2 above)
Comment (avowkind) [reply server/api/person/person.email.js]: it is

**Pull request merged**